src/Tools/dotnet-monitor/Egress/Configuration/EgressConfigurationProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress.Configuration
+{
+    /// <summary>
+    /// Provides access to the Egress:{ProviderType} sections of the configuration.
+    /// </summary>
+    internal sealed class EgressConfigurationProvider :
+        IEgressConfigurationProvider
+    {
+        private readonly IConfigurationSection _egressSection;
+        private readonly IConfigurationSection _propertiesSection;
+
+        public EgressConfigurationProvider(IConfiguration configuration)
+        {
+            _egressSection = configuration.GetEgressSection();
+            _propertiesSection = configuration.GetEgressPropertiesSection();
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<string> ProviderTypes
+        {
+            get
+            {
+                return GetProviderSections().Select(s => s.Key);
+            }
+        }
+
+        /// <inheritdoc/>
+        public IDictionary<string, string> GetAllProperties()
+        {
+            Dictionary<string, string> properties = new Dictionary<string, string>();
+            foreach (IConfigurationSection section in _propertiesSection.GetChildren())
+            {
+                properties.Add(section.Key, section.Value);
+            }
+            return properties;
+        }
+
+        public IConfigurationSection GetProviderConfigurationSection(string providerType, string providerName)
+        {
+            IConfigurationSection section = GetProviderTypeConfigurationSection(providerType).GetSection(providerName);
+
+            if (!section.Exists())
+            {
+                throw new EgressException(string.Format(CultureInfo.CurrentCulture, Strings.ErrorMessage_EgressProviderDoesNotExist, providerName));
+            }
+
+            return section;
+        }
+
+        /// <inheritdoc/>
+        public IConfigurationSection GetProviderTypeConfigurationSection(string providerType)
+        {
+            return GetProviderSections().First(s => s.Key == providerType);
```

Comment:
Do we need to worry about this throwing if there aren't any matching provider sections?